source/content/visual-studio-code.md

```diff
@@ -0,0 +1,141 @@
+
+  ![The Open folder option from the Visual Studio Code Start screen](../images/vscode-open-folder.png)
+
+   - If you have already closed the Welcome screen, click **Open...** in the **File** menu.
+
+1. Navigate your filesystem and select the directory containing your site code (e.g, `~/sites/mysite`).
+1. Save this as a Workspace by selecting **Save Workspace As...** from the **File** menu.
+
+Workspace settings are stored in a file called `.code-workspace` at the root of your project. It is recommended to keep this file out of Git by adding it to your `.gitignore` file at either the site or global level:
+
+```git:title=.gitignore
+// highlight-start
+# Ignore VSCode Workspace Files
+*.code-workspace
+// highlight-end
+
+# WordPress #
+############
+wp-config-local.php
+wp-cli.local.yml
+wp-content/uploads
+wp-content/blogs.dir/
+wp-content/upgrade/
+
+```
+
+At this point, you should see your site files in the File Explorer on the left side of VS Code. Click the file name to open it for editing in the main window.
+
+## Deploy local code changes to a Pantheon environment
+
+Once you've made some code changes, you'll want to push those up to your Pantheon Dev or Multidev environment. You can do this using Git, or with SFTP.
+
+### Push changes with Git (recommended)
+VS Code includes Git integration and an integrated terminal, either of which can be used to make Git changes and push them back to your Pantheon repository and Dev environment.
+
+Make sure your site is in [Git mode](/guides/quickstart/connection-modes) before pushing code via Git.
+
+#### [Git Version Control in VS Code](https://code.visualstudio.com/docs/introvideos/versioncontrol)
+Open the Git menu (**Ctrl** + **Shift** + **G** or **View** -> **SCM**). From here you can review your outstanding changes, commit them with a message, switch branches and more. See the [video on the VS Code docs](https://code.visualstudio.com/docs/introvideos/versioncontrol) for more information.
+
+#### [Integrated Terminal](https://code.visualstudio.com/docs/editor/integrated-terminal)
+Open the integrated terminal with **Ctrl** + **`** or **View** -> **Terminal**. Now you can interact with Git as you would normally, without leaving VS Code.
+
+### Upload your changes with SFTP
+The SFTP extension for VS Code allows developers to upload code to the Pantheon Dev or Multidev environment directly from VS Code, as well as download files from the Pantheon servers.
+
+Make sure your site is in [SFTP mode](/sftp#sftp-mode) before uploading code via SFTP.
+
+<Alert type="info" title="Note">
+
+Due to the structure of the Pantheon filesystem, you'll want to nest your site in a directory named `code` in order for code uploads and downloads to land in the right place on the remote server without hard-coding the full remote server path:
+
+```bash{outputLines:4}
+cd ~/sites/mysite
+mkdir code
+mv ./* .git code
+mv: cannot move './code' to a subdirectory of itself, 'code/code'
```

Comment:
this part threw me off - are we showing people what it looks like when it doesn't work? If so, let's omit the copy-able code.

Reply:
Agreed this part is a little awkward. It's weirdly a little tricky to move all files (including hidden files) into a new directory within the current directory in a way that won't throw warnings. This command works, but it does throw that error (which can be ignored but is still potentially confusing).

This example will leave behind other hidden files as well (e.g, .lando.yml, .htaccess), so this needs updating regardless.

Two ideas here:
- Just go with this, but adjust line 78 to `mv * .* code/` (which will still throw a warning, but works and will grab the rest of the dot files)
- Instead of guiding people clone their site down to `~/sites/mysite` back up in Step 2 in "Before you Begin," we could instruct people to clone their code down to `/sites/mysite/code` from the beginning, which avoids this step entirely (but we'd probably want to briefly note why the "code" subdir is useful)

I'm leaning towards the second option here. Since these directions are geared towards people using SFTP mode, they're less likely to be using the CLI in the first place. What do you think @EdwardAngert?

Reply:
I'm fine with either solution - FWIW the first option could be solved with a `#` comment on that line, but may not be parsed correctly by the more... novice user.

If we reach consensus on the other outstanding comments, I can make the changes needed.

Reply:
I went with option 2 here (https://github.com/pantheon-systems/documentation/commit/d06cde295757bcce25f0ee4ff81abce84c015fbb). I think it's still sort of awkward/hack-y, but at least the COS project will make this sort of thing a non-issue in the near future.